hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -260,6 +260,18 @@ public class HoodieTableConfig extends HoodieConfig {
       .sinceVersion("0.13.0")
       .withDocumentation("The metadata of secondary indexes");
 
+  public static final ConfigProperty<String> AUTO_GENERATE_RECORD_KEYS = ConfigProperty
+      .key("hoodie.table.auto.generate.record.keys")
+      .defaultValue("false")
+      .withDocumentation("Enables automatic generation of the record-keys in cases when dataset bears "
+          + "no natural record key satisfying requirements of being the primary-key in the Hudi table. "
+          + "Record key auto-gen is generally recommended for 'append-only' workloads, ie ones leveraging 'insert' or "
```

Comment:
sure. will take a stab at removing as a table property. 
just that table properties are immutable, where as writer properties are mutable. Since for a table where auto generation is enabled, it does not makes sense to go back to other key gens, I thought it makes sense to go w/ table property.

Reply:
just so we are on same page. your suggestion is to make this a writer config, but still make it immutable? i.e. once set, use can never unset? 
I feel this bit contradicting to how write configs behaves which can keep changing from commit to another. let me know wdyt. I don't have a strong opinion.